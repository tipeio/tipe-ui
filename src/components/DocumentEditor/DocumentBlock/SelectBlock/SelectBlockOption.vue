<template>
  <button
    :data-tipe-ui="$options.name"
    @click="$emit('click', label)"
  >
    <div class="layout">
      <div class="icon-container">
        <div class="icon-box">
          <tipe-icon
            :icon="icon"
            height="1.5rem"
            width="1.5rem"
          />
        </div>
      </div>
      <div class="label">{{ label }}</div>
      <div class="description">{{ description }}</div>
    </div>
  </button>
</template>

<script>
import blockOptionShape from '@/types/BlockOption'
import TipeIcon from '@/components/Icon'

export default {
  name: 'TipeSelectBlockOption',
  components: {
    TipeIcon
  },
  props: blockOptionShape
}
</script>

<style lang="postcss" scoped>
[data-tipe-ui='TipeSelectBlockOption'] {
  display: flex;
  flex-direction: row;
  height: 3.375rem;
  width: 100%;
  margin: 0;
  padding: 0;
  border: none;
  outline: none;
  background-color: transparent;
}

.layout {
  display: grid;
  grid-template-areas:
    'icon label'
    'icon description';
  grid-template-row: 1fr 1fr;
  grid-auto-columns: 3.375rem 1fr;
  height: 3.375rem;
}

.icon-container {
  grid-area: icon;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.icon-box {
  height: 2.625rem;
  width: 2.625rem;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgba(31, 52, 108, 0.05);
  border-radius: 0.25rem;
}

.label {
  grid-area: label;
  display: flex;
  align-items: flex-end;
  justify-content: flex-start;
  color: var(--dark-purple);
  font-size: 0.875rem;
  white-space: nowrap;
  overflow: hidden;
}

.description {
  grid-area: description;
  font-size: 0.75rem;
  color: var(--midPurple);
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  white-space: nowrap;
  overflow: hidden;
}
</style>
