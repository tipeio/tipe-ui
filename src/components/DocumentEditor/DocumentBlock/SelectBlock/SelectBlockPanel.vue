<template>
  <div :data-tipe-ui="$options.name">
    <tipe-select-block-option
      v-for="(option, i) in options"
      v-bind="option"
      :key="i"
      @click="$emit('change', option.type)"
    />
  </div>
</template>

<script>
import vueTypes from 'vue-types'
import blockOptionShape from '@/types/BlockOption'
import TipeSelectBlockOption from './SelectBlockOption'

export default {
  name: 'TipeSelectBlockPanel',
  components: {
    TipeSelectBlockOption
  },
  props: {
    options: vueTypes.arrayOf(vueTypes.shape(blockOptionShape))
  }
}
</script>

<style lang="postcss" scoped>
[data-tipe-ui='TipeSelectBlockPanel'] {
  display: flex;
  flex-direction: column;
  width: 17.5rem;
  background-color: #fff;
  box-shadow: 0 0.125rem 3.375rem 0 rgba(0, 0, 0, 0.2);
  border-radius: 4px;
  padding: 0.9375rem 0.8125rem;
  margin-top: 0.9375rem;
  box-sizing: border-box;
}
</style>
