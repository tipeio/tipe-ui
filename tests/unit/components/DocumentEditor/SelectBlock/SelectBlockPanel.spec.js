import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import SelectBlockOption from '@/components/DocumentEditor/DocumentBlock/SelectBlock/SelectBlockOption'
import SelectBlockPanel from '@/components/DocumentEditor/DocumentBlock/SelectBlock/SelectBlockPanel'
import blockOptionMock from '@/mocks/BlockOption'
import createManyMocks from '@/mocks/createManyMocks'

describe('SelectBlockPanel', () => {
  describe('<template>', () => {
    it('matches previous snapshot', () => {
      const renderer = createRenderer()
      const wrapper = shallowMount(SelectBlockPanel)

      renderer.renderToString(wrapper.vm, (err, str) => {
        if (err) throw new Error(err)
        expect(str).toMatchSnapshot()
      })
    })

    it('has correct data-tipe-iu attibute', () => {
      const wrapper = shallowMount(SelectBlockPanel)

      expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeSelectBlockPanel')
    })
  })

  describe(':props', () => {
    it(':options - should render options', () => {
      const propsData = { options: createManyMocks(blockOptionMock, 3) }
      const wrapper = shallowMount(SelectBlockPanel, { propsData })

      expect(wrapper.findAll(SelectBlockOption)).toHaveLength(3)
    })

    it(':options.icon - should pass the SelectBlockOption component as icon prop', () => {
      const propsData = { options: createManyMocks(blockOptionMock, 1) }
      const wrapper = shallowMount(SelectBlockPanel, { propsData })

      expect(wrapper.find(SelectBlockOption).props().icon).toEqual(
        propsData.options[0].icon
      )
    })

    it(':options.label - should pass the SelectBlockOption component as label prop', () => {
      const propsData = { options: createManyMocks(blockOptionMock, 1) }
      const wrapper = shallowMount(SelectBlockPanel, { propsData })

      expect(wrapper.find(SelectBlockOption).props().label).toEqual(
        propsData.options[0].label
      )
    })

    it(':options.description - should pass the SelectBlockOption component as description prop', () => {
      const propsData = { options: createManyMocks(blockOptionMock, 1) }
      const wrapper = shallowMount(SelectBlockPanel, { propsData })

      expect(wrapper.find(SelectBlockOption).props().description).toEqual(
        propsData.options[0].description
      )
    })
  })
})
