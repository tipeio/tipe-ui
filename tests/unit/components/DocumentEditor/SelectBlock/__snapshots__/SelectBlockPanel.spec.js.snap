// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBlockPanel <template> matches previous snapshot 1`] = `<div data-tipe-ui="TipeSelectBlockPanel"></div>`;
